Bug: Test regression in `test_requirement_source_require_hashes_unpinned`

### Pre-submission checks

- [x] I am **not** filing an auditing error (false positive or negative). These **must** be reported to [pypa/advisory-database](https://github.com/pypa/advisory-database/issues/new) instead.
- [x] I agree to follow the [PSF Code of Conduct](https://www.python.org/psf/conduct/).
- [x] I have looked through the open issues for a duplicate report.

### Expected behavior

All tests pass

### Actual behavior

The `test_requirement_source_require_hashes_unpinned` test fails.

```
=================================== FAILURES ===================================
_______________ test_requirement_source_require_hashes_unpinned ________________

req_file = <function req_file.<locals>._req_file at 0x7f38e0d89380>

    def test_requirement_source_require_hashes_unpinned(req_file):
        source = _init_requirement(
            [
                (
                    req_file(),
                    "wheel==0.38.1 "
                    "--hash=sha256:7a95f9a8dc0924ef318bd55b616112c70903192f524d120acc614f59547a9e1f\n"
                    "setuptools<=67.0.0 "
                    "--hash=sha256:9d790961ba6219e9ff7d9557622d2fe136816a264dd01d5997cfc057d804853d",
                )
            ]
        )
    
        # When hashed dependencies are provided, all dependencies must be explicitly pinned to an exact
        # version number
>       with pytest.raises(DependencySourceError):
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       Failed: DID NOT RAISE <class 'pip_audit._dependency_source.interface.DependencySourceError'>

test/dependency_source/test_requirement.py:490: Failed
=============================== warnings summary ===============================
env/lib/python3.14/site-packages/pip_api/_vendor/pyparsing.py:108
  /home/runner/work/pip-audit/pip-audit/env/lib/python3.14/site-packages/pip_api/_vendor/pyparsing.py:108: DeprecationWarning: module 'sre_constants' is deprecated
    import sre_constants

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/dependency_source/test_requirement.py::test_requirement_source_require_hashes_unpinned - Failed: DID NOT RAISE <class 'pip_audit._dependency_source.interface.DependencySourceError'>
============= 1 failed, 237 passed, 1 warning in 127.20s (0:02:07) =============
make: *** [Makefile:66: test] Error 1
```

### Reproduction steps

Run the test suite.

### Logs

```plain text

```

### Additional context

_No response_

### OS name, version, and architecture

_No response_

### pip-audit version

HEAD

### pip version

latest

### Python version

all

--- a/test/dependency_source/test_requirement.py
+++ b/test/dependency_source/test_requirement.py
@@ -477,6 +477,7 @@ def test_requirement_source_require_hashes_unpinned(req_file):
         [
             (
                 req_file(),
+                "missing\n"
                 "wheel==0.38.1 "
                 "--hash=sha256:7a95f9a8dc0924ef318bd55b616112c70903192f524d120acc614f59547a9e1f\n"
                 "setuptools<=67.0.0 "

--- a/test/dependency_source/test_requirement.py
+++ b/test/dependency_source/test_requirement.py
@@ -536,7 +536,9 @@ def test_requirement_source_disable_pip_duplicate_dependencies(req_file):
 
 def test_requirement_source_disable_pip_duplicate_dependencies_with_extras(req_file):
     source = _init_requirement(
-        [(req_file(), "aiohttp==3.9\naiohttp[speedups]==3.9")], disable_pip=True, no_deps=True
+        [(req_file(), "aiohttp==3.9\naiohttp[speedups]==3.9")],
+        disable_pip=True,
+        no_deps=True,
     )
 
     specs = list(source.collect())

